OpenEdXMobile/src/main/java/org/edx/mobile/util/CourseDateUtil.kt

```diff
@@ -46,29 +46,49 @@ object CourseDateUtil {
             title.visibility = View.VISIBLE
             containerLayout.setBackgroundColor(Color.TRANSPARENT)
         }
-        when (bannerType) {
-            CourseBannerType.UPGRADE_TO_GRADED -> {
-                bannerMessage.text = context.getText(if (isCourseDatePage) R.string.course_dates_banner_upgrade_to_graded else R.string.course_dashboard_banner_upgrade_to_graded)
-                biValue = Analytics.Values.COURSE_DATES_BANNER_UPGRADE_TO_PARTICIPATE
-                bannerTypeValue = Analytics.Values.PLS_BANNER_TYPE_UPGRADE_TO_PARTICIPATE
-            }
-            CourseBannerType.UPGRADE_TO_RESET -> {
-                bannerMessage.text = context.getText(if (isCourseDatePage) R.string.course_dates_banner_upgrade_to_reset else R.string.course_dashboard_banner_upgrade_to_reset)
-                biValue = Analytics.Values.COURSE_DATES_BANNER_UPGRADE_TO_SHIFT
-                bannerTypeValue = Analytics.Values.PLS_BANNER_TYPE_UPGRADE_TO_SHIFT
-            }
-            CourseBannerType.RESET_DATES -> {
-                bannerMessage.text = context.getText(if (isCourseDatePage) R.string.course_dates_banner_reset_date else R.string.course_dashboard_banner_reset_date)
-                buttonText = context.getString(R.string.course_dates_banner_reset_date_button)
-                biValue = Analytics.Values.COURSE_DATES_BANNER_SHIFT_DATES
-                bannerTypeValue = Analytics.Values.PLS_BANNER_TYPE_SHIFT_DATES
+
```

Comment:
IMO, better to define a new variable for BannerDescription and mark the check on that string variable as we are setting the default value in XML to `bannerMessage` and it will always be a non-empty string when we will check it on [line:94](https://github.com/openedx/edx-app-android/pull/1756/files#diff-3959f2aeefd5f800f77204bfc983f5eecf4b1d7bcc8ed80fa8868860e9742338R94).

Reply:
we can re-write this code as
```
        val context = view.context as Context
        val containerLayout = view as LinearLayout
        val title = view.findViewById(R.id.banner_title) as TextView
        val bannerMessage = view.findViewById(R.id.banner_info) as TextView
        val imgView = view.findViewById(R.id.iv_calender) as ImageView
        val button = view.findViewById(R.id.btn_shift_dates) as Button
        var description: CharSequence = ""
        var buttonText = ""
        var bannerTypeValue = ""
        var biValue = ""
        val bannerType: CourseBannerType =
            courseBannerInfoModel.datesBannerInfo.getCourseBannerType()
        // Currently we are only handling RESET_DATES case,
        // TODO UPGRADE_TO_GRADED & UPGRADE_TO_RESET will be enable once we are allowed to do payment through mobile
        if (isCourseDatePage) {
            title.visibility = View.VISIBLE
            containerLayout.setBackgroundColor(Color.TRANSPARENT)
        }

        // CourseBannerType.UPGRADE_TO_GRADED && CourseBannerType.UPGRADE_TO_RESET will only be handled for CourseDatesPage
        when (bannerType) {
            CourseBannerType.UPGRADE_TO_GRADED -> {
                if (isCourseDatePage) {
                    description = context.getText(R.string.course_dates_banner_upgrade_to_graded)
                    biValue = Analytics.Values.COURSE_DATES_BANNER_UPGRADE_TO_PARTICIPATE
                    bannerTypeValue = Analytics.Values.PLS_BANNER_TYPE_UPGRADE_TO_PARTICIPATE
                }
            }
            CourseBannerType.UPGRADE_TO_RESET -> {
                if (isCourseDatePage) {
                    description = context.getText(R.string.course_dates_banner_upgrade_to_reset)
                    biValue = Analytics.Values.COURSE_DATES_BANNER_UPGRADE_TO_SHIFT
                    bannerTypeValue = Analytics.Values.PLS_BANNER_TYPE_UPGRADE_TO_SHIFT
                }
            }
            CourseBannerType.RESET_DATES -> {
                description = context.getText(
                    if (isCourseDatePage) R.string.course_dates_banner_reset_date
                    else R.string.course_dashboard_banner_reset_date
                )
                buttonText = context.getString(R.string.course_dates_banner_reset_date_button)
                biValue = Analytics.Values.COURSE_DATES_BANNER_SHIFT_DATES
                bannerTypeValue = Analytics.Values.PLS_BANNER_TYPE_SHIFT_DATES
            }
            CourseBannerType.INFO_BANNER -> {
                description = context.getText(
                    if (isCourseDatePage) R.string.course_dates_info_banner
                    else R.string.course_dashboard_info_banner
                )
                biValue = Analytics.Values.COURSE_DATES_BANNER_INFO
                bannerTypeValue = Analytics.Values.PLS_BANNER_TYPE_INFO
            }
            CourseBannerType.BLANK -> description = ""
        }

        if (!TextUtils.isEmpty(description) && (isSelfPaced || (isSelfPaced.not() && bannerType == CourseBannerType.UPGRADE_TO_GRADED))) {
            bannerMessage.text = description
            if (!TextUtils.isEmpty(buttonText)) {
                button.text = buttonText
                button.visibility = View.VISIBLE
                imgView.visibility = if (isCourseDatePage) View.GONE else View.VISIBLE
                button.setOnClickListener { v ->
                    clickListener.onClick(v)
                    analyticsRegistry.trackPLSShiftButtonTapped(courseId, enrollmentMode, screenName)
                }
            } else {
                button.visibility = View.GONE
                imgView.visibility = View.GONE
            }
            view.visibility = View.VISIBLE
            analyticsRegistry.trackPLSCourseDatesBanner(biValue, courseId, enrollmentMode, screenName, bannerTypeValue)
        } else {
            view.visibility = View.GONE
        }